utils/grass_clang_format.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/sh
+set -eu
+
+###############################################################################
+# Format source code according to GRASS GIS submitting rules
+#
+# Dependencies:
+#    clang-format
+#      (most easily available with e.g.:
+#      `python -m pip install 'clang-format==15.0.6’`)
+#
+# Author(s):
+#    Nicklas Larsson
+#
+# Usage:
+#    If you have "clang-format" in PATH, execute for complete source formatting:
+#      ./utils/grass_clang_format.sh
+#
+#    Setting 'GRASS_CLANG_FORMAT' to explicitly set clang-format version:
+#      GRASS_CLANG_FORMAT="clang-format-15" ./utils/grass_clang_format.sh
+#
+#    It is also possible to format the content in a (one) given directory:
+#      ./utils/grass_clang_format.sh ./lib/raster
+#
+# COPYRIGHT: (C) 2023 by the GRASS Development Team
+#
+#            This program is free software under the GNU General Public
+#            License (>=v2). Read the file COPYING that comes with GRASS
+#            for details.
+#
+###############################################################################
+
+# Required clang-format version
+req_cf_v="15"
+
+# No need to continue if the .clang-format file isn't found
+if [ ! -f .clang-format ] && [ ! -f ../.clang-format ]; then
```

Comment:
```suggestion
if [ ! -f .clang-format ] && [ ! -f ../.clang-format ]; then
```

While we have our official file in place at https://github.com/OSGeo/grass/blob/main/.clang-format (resp. in a local repo copy thru `git clone`) we cannot be sure that `../.clang-format` contains the same.
Perhaps better limit the choice to the file included in the repo?

Reply:
You're right. I made a logical mismatch, figured this to be good for rebase/merging old PRs to main, but then you need this script too.... ahhhh. Will update to check only in source dir.